data/pigui/modules/new-game-window/recovery.lua

```diff
@@ -0,0 +1,43 @@
+-- Copyright © 2008-2024 Pioneer Developers. See AUTHORS.txt for details
+-- Licensed under the terms of the GPL v3. See licenses/GPL-3.txt
+
+local Json = require 'Json'
+local Lang = require 'Lang'
+local lui = Lang.GetResource("ui-core")
+
+local Recovery = {}
+
+-- patch
+--    from - number (version)
+--    to - number (version)
+--    apply(gameDoc) -> errorString? - function that modifies whole gameDoc
+--
+--    it is assumed that the patches are arranged in the array in ascending
+--    order of versions, and do not intersect
+local Patches = {}
```

Comment:
This should probably be made public, as I'd want to write savefile migration functions in their own file which `require()s` the `Recovery` module.

Reply:
Done!